cli/cli.py

```diff
@@ -0,0 +1,282 @@
+        return self.data.get("parameters", {})
+
+    def _is_singular(self):
+        if self.method == "post":
+            # /media/permissions or /collections/permissions
+            if self.path.endswith("permissions"):
+                return False
+
+            return True
+
+        if self.method != "get":
+            return False
+
+        # "/media/{media_id}/download": {
+        if "200" not in self.data["responses"]:
+            return False
+
+        return_ok = self.data["responses"]["200"]
+
+        # /perspectives/{perspective_id}/insights/overview
+        if "schema" not in return_ok:
+            return False
+
+        #     "/media/search": {
+        schema = return_ok["schema"]
+        props = schema.get("properties", None) or schema["allOf"][0].get(
+            "properties", None
+        )
+
+        # "/collections/{collection_id}": {
+        return_types = [value.get("type", "") for _, value in props.items()]
+
+        return any([return_type != "array" for return_type in return_types])
+
+    def _create_command_name(self, method, path_entities):
+        if path_entities[-1] in [
+            "search",
+            "download",
+            "complete",
+            "ping",
+            "transfer_media",
+        ]:
+            return "_".join([path_entities[-1]] + path_entities[:-1])
+        else:
+            command_name = (
+                f"{Command._method_to_command[method]}_{'_'.join(path_entities)}"
+            )
+            if command_name[-1] == "s" and self._is_singular():
+                return command_name[:-1]
+            else:
+                return command_name
+
+    def _is_csv_command(self):
+        return self.name in [
+            "show_perspectives_insights_overview",
+            "show_perspectives_insights_users",
+        ]
+
+    def _process_response(self, response, args):
+        # can be 'application/json; charset=utf-8' or 'video/mp4'
+        content_type = response.headers["Content-Type"].split(";")
+        if content_type[0] == "application/json":
+            return json.dumps(response.json(), indent=2)
+
+        if self.name.startswith("download_"):
+            if content_type[0] in ["video/mp4", "audio/mp4"]:
+                extension = "mp4"
+            else:
+                # for captions it's text/plain
+                extension = "str"
+
+            output_filename = f"{str(uuid.uuid4())}.{extension}"
+            with open(output_filename, "wb") as tf:
+                tf.write(response.content)
+
+            return f"Downloaded {content_type} content to {output_filename}"
+
+        if self._is_csv_command() and args.table_format:
+            result = []
+
+            csv_string = response.content.decode()
+            reader = csv.reader(io.StringIO(csv_string))
+
+            header = next(reader)
+            for row in reader:
+                result.append(row)
+
+            # https://github.com/astanin/python-tabulate#table-format
```

Comment:
I would probably also remove this comment.

Reply:
👍